engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -367,39 +392,80 @@ public boolean modifyRowKeys(final SingletonContext context, @NotNull final Long
         }
         // noinspection unchecked
         final LongChunk<OrderedRowKeys> rowKeysAsOrdered = (LongChunk<OrderedRowKeys>) rowKeys;
-        accumulateToRowSet(modifiedRowSets, rowKeysAsOrdered, 0, rowKeys.size(), destination);
+        accumulateToBuilderRandom(modifiedBuilders, rowKeysAsOrdered, 0, rowKeys.size(), destination);
         stepUpdatedDestinations.insert(destination);
         return false;
     }
 
-    private static void accumulateToRowSet(@NotNull final ObjectArraySource<WritableRowSet> rowSetColumn,
+    private static void accumulateToBuilderSequential(
+            @NotNull final ObjectArraySource<RowSetBuilderSequential> rowSetColumn,
             @NotNull final LongChunk<OrderedRowKeys> rowKeysToAdd,
             final int start, final int length, final long destination) {
-        final WritableRowSet rowSet = rowSetColumn.getUnsafe(destination);
-        if (rowSet == NONEXISTENT_TABLE_ROW_SET) {
+        final RowSetBuilderSequential builder = rowSetColumn.getUnsafe(destination);
+        // slice the chunk to the start and length
+        LongChunk<OrderedRowKeys> sliced = rowKeysToAdd.slice(start, length);
```

Comment:
I don't particularly like allocating a sliced view for every add. I would have one `ResettableLongChunk` stored as an instance member and re-use it here and in the other method that slices. Be sure to allocate using `io.deephaven.chunk.ResettableLongChunk#makeResettableChunkForPool` to avoid complaints about chunk leaks in the unit test instrumentation.

Irrelevant style note: this code is pedantic about using `final` wherever possible. It scratches some itch in my brain.